Sources/Apollo/InMemoryNormalizedCache.swift

```diff
@@ -4,16 +4,14 @@ public final class InMemoryNormalizedCache: NormalizedCache {
   private var records: RecordSet
   private let recordsLock = NSRecursiveLock()
 
-  public init(records: RecordSet = RecordSet()) {
+  public init(records: RecordSet = .init()) {
     self.records = records
   }
 
   public func loadRecords(forKeys keys: [CacheKey],
                           callbackQueue: DispatchQueue?,
                           completion: @escaping (Result<[Record?], Error>) -> Void) {
-    self.recordsLock.lock()
-    let records = keys.map { self.records[$0] }
-    self.recordsLock.unlock()
+    let records = self.threadSafe { keys.map{ self.records[$0] } }
```

Comment:
@martijnwalraven I think there was some point at which you were planning on removing the locks here - what's your take on whether we should keep them or not?

Reply:
We should definitely remove those locks (and from `InMemoryNormalizedCache` as well). They serve no real useful purpose, because transactions only exist at the level of the store, and that's the proper level of isolation. You should be able to share a store between clients, but a cache should never be shared.

Reply:
Removed